lib/components/viewers/route-details.js

```diff
@@ -0,0 +1,214 @@
+      patterns &&
+      Object.entries(patterns)
+        .map((pattern) => {
+          return {
+            geometryLength: pattern[1].geometry?.length,
+            headsign: extractHeadsignFromPattern(pattern[1]),
+            id: pattern[0]
+          }
+        })
+        // Remove duplicate headsigns. Using a reducer means that the first pattern
+        // with a specific headsign is the accepted one. TODO: is this good behavior?
+        .reduce((prev, cur) => {
+          const amended = prev
+          const alreadyExistingIndex = prev.findIndex(
+            (h) => h.headsign === cur.headsign
+          )
+          // If the item we're replacing has less geometry, replace it!
+          if (alreadyExistingIndex >= 0) {
+            // Only replace if new pattern has greater geometry
+            if (
+              amended[alreadyExistingIndex].geometryLength < cur.geometryLength
+            ) {
+              amended[alreadyExistingIndex] = cur
+            }
+          } else {
+            amended.push(cur)
+          }
+          return amended
+        }, [])
+        .sort((a, b) => {
+          // sort by number of vehicles on that pattern
+          const aVehicleCount = route.vehicles?.filter(
+            (vehicle) => vehicle.patternId === a.id
+          ).length
+          const bVehicleCount = route.vehicles?.filter(
+            (vehicle) => vehicle.patternId === b.id
+          ).length
+
+          // if both have the same count, sort by pattern geometry length
+          if (aVehicleCount === bVehicleCount) {
+            return b.geometryLength - a.geometryLength
+          }
+          return bVehicleCount - aVehicleCount
+        })
+
+    // if no pattern is set, we are in the routeRow
+    return (
+      <Container full={pattern != null}>
+        <RouteNameContainer>
+          <LogoLinkContainer>
+            {agency && <FormattedMessage
+              id='components.RouteDetails.operatedBy'
+              values={{
+                agencyName: agency.name
+              }}
+            />}
+            {url && (
+              <a href={url} target='_blank'>
+                <Icon type='link' />
+                <FormattedMessage id='components.RouteDetails.moreDetails' />
+              </a>
+            )}
+          </LogoLinkContainer>
+        </RouteNameContainer>
+        <PatternContainer>
+          <h4>
+            <label htmlFor='headsign-selector' id='headsign-selector-label'>
+              <FormattedMessage id='components.RouteDetails.stopsTo' />
+            </label>
+          </h4>
+          {headsigns &&
+          <select
+            id='headsign-selector'
+            name='headsigns'
+            onBlur={this._headSignButtonClicked}
+            onChange={this._headSignButtonClicked}
+            value={viewedRoute?.patternId}
+          >
+            {!viewedRoute.patternId && <option>{intl.formatMessage({id: 'components.RouteDetails.selectADirection'})}</option>}
```

Comment:
`viewedRoute` might be briefly null; add a null check.

Reply:
Good catch thank you! Fixed in 092b53f and checked all files for other problematic references.